test/specs.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+var SearchParser = require('../SearchParser');
+var expect = require('chai').expect;
+
+describe("search-parser", function () {
+    var testPatterns = [
+        {
+            param: 'param1',
+            pattern: /^([a-z]_[0-9]{1,5})$/
+        },
+        {
+            param: 'param2',
+            pattern: /^([a-z]_[0-9]{6,}(?:,\s*[a-z]_[0-9]{6,})*)$/
+        },
+        {
+            param: 'param3',
+            pattern: /^([0-9]{6,}(?:,\s*[0-9]{6,})*)$/
+        },
+        {
+            pattern: /^(.+?)\s(.+\s)?(.+)|(.*)$/i,
+            parse: function (result) {
+                return {
+                    subParam1: result[1],
+                    subParam2: result[3] || result[4]
+                };
+            }
+        }
+    ];
```

Comment:
Not a blocker, but it's better to re-define your dest data inside of the beforeEach to ensure that it doesn't get corrupted between runs. Not a blocker, since what you're doing here is pretty straight-forwrd.